src/gccrs/args.rs

```diff
@@ -0,0 +1,152 @@
+    /// Remaining options, handled by `rustc` but not `gccrs`
+    Unknown,
+}
+
+impl CrateType {
+    /// Get the corresponding [`CrateType`] from a given option to the `--crate-type`
+    /// option
+    pub fn from_str(s: &str) -> CrateType {
+        match s {
+            "bin" => CrateType::Bin,
+            "dylib" => CrateType::DyLib,
+            "staticlib" => CrateType::StaticLib,
+            _ => CrateType::Unknown,
+        }
+    }
+}
+
+/// Structure used to represent arguments passed to `gccrs`. Convert them from `rustc`
+/// arguments using [`GccrsArg::from_rustc_arg`]
+pub struct GccrsArgs {
+    source_files: Vec<String>,
+    crate_type: CrateType,
+    output_file: String,
+}
+
+impl GccrsArgs {
+    fn generate_parser() -> Options {
+        let mut options = Options::new();
+        options.optopt("", "crate-name", "Name of the crate to compile", "NAME");
+        options.optopt("", "edition", "Rust edition to use", "YEAR");
+        options.optopt("", "error-format", "Requested error format", "EXTENSION");
+        options.optopt(
+            "",
+            "out-dir",
+            "Directory in which to output generated files",
+            "DIR",
+        );
+        options.optopt("", "emit", "Requested output to emit", "KIND");
+        options.optopt("", "json", "JSON Rendering type", "RENDER");
+        options.optopt("", "json", "JSON Rendering type", "RENDER");
+        options.optmulti("C", "", "Extra compiler options", "OPTION[=VALUE]");
+        options.optmulti(
+            "L",
+            "",
+            "Add a directory to the library's search path",
+            "KIND[=PATH]",
+        );
+        options.optmulti("", "crate-type", "Type of binary to output", "TYPE");
+
+        options
+    }
+
+    fn format_output_filename(matches: &Matches, crate_type: &CrateType) -> String {
+        // FIXME: No unwraps here
+        let crate_name = matches.opt_str("crate-name").unwrap();
+        let out_dir = matches.opt_str("out-dir").unwrap();
+
+        // FIXME: Figure out a way to return multiple output filenames. Just handle `bin`
+        // for now
+        let c_options = matches.opt_strs("C");
+
+        // FIXME: This is probably different on Windows, we should use Paths and PathStrs
+        // instead
+        let mut output_file = match crate_type {
+            &CrateType::Bin => format!("{}/{}", out_dir, crate_name),
+            &CrateType::DyLib => format!("{}/lib{}.so", out_dir, crate_name),
+            &CrateType::StaticLib => format!("{}/lib{}.a", out_dir, crate_name),
+            _ => unreachable!(
+                "gccrs cannot handle other crate types than bin, dylib or staticlib at the moment"
+            ),
+        };
+
+        // FIXME: Horrendous. We need to create a separate "C options" parser since we'll
+        // probably use more than just `extra-filename`.
+        c_options
+            .iter()
+            .for_each(|c_opt| match c_opt.split('=').nth(0).as_deref() {
```

Comment:
I wouldn't use `for_each` here. Just use a normal for loop.

Reply:
This is horrendous code anyway, there are two calls to `.split('=')`, etc. I wanted to get a starting point on parsing `-C` options, since we only need `extra-filename` for now. It's gonna get split up somewhere different.